tests/examples/common.py

```diff
@@ -0,0 +1,63 @@
+import os
+import tempfile
+import subprocess
+
+MOCOCRW_EXAMPLE_BINARY_PATH = "."
+try:
+    MOCOCRW_EXAMPLE_BINARY_PATH = os.environ["MOCOCRW_EXAMPLE_BINARY_PATH"]
+except KeyError:
+    print("Please set the MOCOCRW_EXAMPLE_BINARY_PATH environment variable.")
+    exit(1)
+
+MOCOCRW_DIR = ""
+try:
+    MOCOCRW_SRC_DIR = os.environ["MOCOCRW_SRC_DIR"]
+except KeyError:
+    print("Please set the MOCOCRW_SRC_DIR environment variable")
+    exit(1)
+
+
+HASH_ALGOS = [
+    "SHA256", "SHA384", "SHA512", "SHA3-256", "SHA3-384", "SHA3-512"
+]
+
+
+def add_invalid_parameters_and_execute(execute_string):
+    def execute(exec_string):
+        result = subprocess.run(exec_string.split(), capture_output=True)
+        assert result.returncode == 1
+        assert b"Unknown option selected: unrecognised option '--aw\xc3\xb6eljf'\n" \
+               b"Failure parsing command line.\n" == result.stderr
+    execute(execute_string + " --awöeljf asdölvkjyxc")
+
+
+def create_temp_working_dir():
+    # Create working directory
+    new_working_dir = tempfile.mkdtemp()
+
+    current_working_dir = os.getcwd()
+    # Change to working directory
+    os.chdir(new_working_dir)
+    return current_working_dir, new_working_dir
+
+
+def combine_all_options(all_options, option_list):
```

Comment:
Please write a documentation comment to explain what this function does, because I can't figure it out from reading the source.